pkg/handlers/update_checker_spec.go

```diff
@@ -146,36 +131,68 @@ func (h *Handler) ConfigureAutomaticUpdates(w http.ResponseWriter, r *http.Reque
 	if cronSpec != "@never" && cronSpec != "@default" {
 		_, err := cron.ParseStandard(cronSpec)
 		if err != nil {
-			logger.Error(err)
 			updateCheckerSpecResponse.Error = "failed to parse cron spec"
-			JSON(w, 500, updateCheckerSpecResponse)
+			logger.Error(errors.Wrap(err, updateCheckerSpecResponse.Error))
+			JSON(w, http.StatusInternalServerError, updateCheckerSpecResponse)
 			return
 		}
 	}
 
-	cm.Data[fmt.Sprintf("update-schedule-%s", foundApp.ID)] = cronSpec
-	store.GetStore().UpdateConfigmap(cm)
-	if err != nil {
-		logger.Error(err)
-		updateCheckerSpecResponse.Error = "failed to update config map"
-		JSON(w, 500, updateCheckerSpecResponse)
+	if err := store.GetStore().SetUpdateCheckerSpec(foundApp.ID, cronSpec); err != nil {
+		updateCheckerSpecResponse.Error = "failed to set update checker spec"
+		logger.Error(errors.Wrap(err, updateCheckerSpecResponse.Error))
+		JSON(w, http.StatusInternalServerError, updateCheckerSpecResponse)
 		return
 	}
 
 	if err := store.GetStore().SetAutoDeploy(foundApp.ID, configureAutomaticUpdatesRequest.AutoDeploy); err != nil {
-		logger.Error(err)
 		updateCheckerSpecResponse.Error = "failed to set auto deploy"
-		JSON(w, 500, updateCheckerSpecResponse)
+		logger.Error(errors.Wrap(err, updateCheckerSpecResponse.Error))
+		JSON(w, http.StatusInternalServerError, updateCheckerSpecResponse)
 		return
 	}
 
 	// reconfigure update checker for the app
-	if err := updatechecker.Configure(foundApp); err != nil {
-		logger.Error(err)
+	if err := updatechecker.Configure(foundApp, cronSpec); err != nil {
 		updateCheckerSpecResponse.Error = "failed to reconfigure update checker cron job"
-		JSON(w, 500, updateCheckerSpecResponse)
+		logger.Error(errors.Wrap(err, updateCheckerSpecResponse.Error))
+		JSON(w, http.StatusInternalServerError, updateCheckerSpecResponse)
 		return
 	}
 
-	JSON(w, 204, "")
+	JSON(w, http.StatusNoContent, "")
+}
+
+func (h *Handler) GetAutomaticUpdatesConfig(w http.ResponseWriter, r *http.Request) {
```

Comment:
this used to be part of the application object returned to the ui when listing apps, is there a reason it's a separate handler now?